### PR TITLE
Small USB section (don't think it should be more)

### PR DESCRIPTION
**Pull request by nillas12**
After mucking up Thomas' branch, my own is now ready. Sorry Thomas.

### CODE REVIEW

File: report/4.Design/USB.tex [hunk 1]
```diff
@@ -0,0 +1,26 @@
+\section{USB communication}
+\label{sec:usb}
+As described in the component diagram in section \ref{sec:architecture}, the system will consist of two seperate hardware platforms, that need to communicate.
+As the NXT is limited in communication to BlueTooth and USB, and bluetooth is too slow for a real-time system, the group has decided to use USB.
```
Comment (frederikspang) [inline report/4.Design/USB.tex]: Inconsistent "bluetooth" and "BlueTooth"

File: report/4.Design/USB.tex [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+According to the NXTOSEK documentation, the NXTOSEK library should also support built-in USB, through an easy to use API.
```
Comment (frederikspang) [inline report/4.Design/USB.tex]: "nxtOSEK" *

File: report/4.Design/USB.tex [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+When the group tested this functionality, it turned out that the driver for the PC/ RaspBerry no longer worked.
+
+However, since the driver is simply meant to be a translator between computers, and the devices attatched to them, the group decided to build their own driver, instead of just doing raw bit communication.
```
Comment (frederikspang) [inline report/4.Design/USB.tex]: "attached"

File: report/4.Design/USB.tex [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+The driver library will consist of three parts.
+The initialization part, which will perform a handshake with the device, telling the NXT that it should expect communication.
+This part posed a bit of a challenge, as it was not documented anywhere what the handshake consisted of, or that it even existed.
```
Comment (frederikspang) [inline report/4.Design/USB.tex]: Consider "This part posed a bit of a challenge, as it was not documented anywhere what the handshake consisted of, or that it even existed."

to 

"This part posed a bit of a challenge, as the handshake and its content was not documented, thus leaving us in the dark about its existence" el. lignede omformulering

File: report/4.Design/USB.tex [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+The group used a USB sniffer, attached to a working compiled program, to determine that the NXT expected two bytes x01, a 1 and xFF, 255.
```
Comment (frederikspang) [inline report/4.Design/USB.tex]: `\texttt{x01}` og even `0x01` to be more "precise".. Same for `xFF` and `255`?

File: report/4.Design/USB.tex [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+The data writing part will perform the actual communication with the device, and will be concerned with sending the actual packets of data.
+
+Finally a disconnect part, which will send a message that tells the NXT to shut off the USB communication.
```
Comment (frederikspang) [inline report/4.Design/USB.tex]: `terminate` over "shut off".. Or "shut down"?

Comment (frederikspang) [reply report/4.Design/USB.tex]: Or even "disconnect"?

File: report/4.Design/USB.tex [hunk 7]
```diff
@@ -0,0 +1,32 @@
+\section{USB communication}
+\label{sec:usb}
+As described in the component diagram in Section \ref{sec:architecture}, the system will consist of two separate hardware platforms, that need to communicate.
```
Comment (Moesgaarda) [inline report/4.Design/USB.tex]:
```suggestion
As described in the component diagram in Section~\ref{sec:architecture}, the system will consist of two separate hardware platforms, that need to communicate.
```

File: report/4.Design/USB.tex [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+As the NXT can only communicate with external devices using Bluetooth and USB, and we deemed Bluetooth too slow for our system, the group has decided to use USB.
+According to the nxtOSEK documentation, the nxtOSEK library should also support built-in USB, through an easy to use API\cite{ecrobotUSB}.
+When the group tested this functionality, it turned out that the nxtOSEK library for the computer, was no longer functional.
+
+The group tried a few different solutions for custom NXT USB communication, but in the end, writing our own module to handle the communication was deemed the best solution.
+
+The communication is described in the diagram here:
```
Comment (Moesgaarda) [inline report/4.Design/USB.tex]:
```suggestion
The communication is described on Figure~\ref{fig:compusb}.:
```

Comment (Moesgaarda) [reply report/4.Design/USB.tex]: eller is visualized

**Pull request merged**